src/kvstore/kvstore_dist.h

```diff
@@ -181,16 +223,21 @@ class KVStoreDist : public KVStoreLocal {
           RunContext rctx, Engine::CallbackOnComplete cb) {
         // convert to ps keys
         size_t size = recv_buf.shape().Size();
-        PSKV& pskv = EncodeKey(key, size);
+
+        // even if inactive should use compressed_pskv for pull when type is not none
+        PSKV& pskv = (gc_->get_type() == GC_NONE) ?
+                      EncodeDefaultKey(key, size, false) :
+                      EncodeCompressedKey(key, size, false);
 #if MKL_EXPERIMENTAL == 1
         mkl_set_tblob_eager_mode(recv_buf.data());
 #endif
         real_t* data = recv_buf.data().dptr<real_t>();
         // false means not to delete data when SArray is deleted
         auto vals = new ps::SArray<real_t>(data, size, false);
         // issue pull
+        int cmd = (gc_->get_type() != GC_NONE) ? kCompressedPushPull : kDefaultPushPull;
```

Comment:
Why not just pass gc->get_type to ZPull?

Reply:
This allows uniformity with existing code. GC_NONE enum value would conflict with other modes in kvdist_server